app/models/manageiq/providers/redhat/infra_manager/vm_or_template_shared/scanning.rb

```diff
@@ -28,8 +28,82 @@ def perform_metadata_sync(ost)
     sync_stashed_metadata(ost)
   end
 
+  def proxies4job(job = nil)
+    _log.debug "Enter (RHEVM)"
+    msg = 'Perform SmartState Analysis on this VM'
+    embedded_msg = nil
+
+    # If we do not get passed an model object assume it is a job guid
+    if job && !job.kind_of?(ActiveRecord::Base)
+      jobid = job
+      job = Job.find_by_guid(jobid)
+    end
+
+    all_proxy_list = storage2proxies
+    proxies = storage2active_proxies(all_proxy_list)
+    _log.debug "# proxies = #{proxies.length}"
+
+    if proxies.empty?
+      msg = embedded_msg.nil? ? 'No active SmartProxies found to analyze this VM' : embedded_msg
+    end
+    log_proxies(proxies, all_proxy_list, msg, job) if proxies.empty? && job
+
+    {:proxies => proxies.flatten, :message => msg}
+  end
+
+  def miq_server_proxies
+    _log.debug "Enter (RHEVM)"
+
+    _log.debug "RedHat: storage_id.blank? = #{storage_id.blank?}"
+    return [] if storage_id.blank?
+
+    storage_server_ids = storages.collect { |s| s.vm_scan_affinity.collect(&:id) }.reject(&:blank?)
+    _log.debug "storage_server_ids.length = #{storage_server_ids.length}"
+
+    all_storage_server_ids = storage_server_ids.inject(:&) || []
+    _log.debug "all_storage_server_ids.length = #{all_storage_server_ids.length}"
+
+    srs = self.class.miq_servers_for_scan
+    _log.debug "srs.length = #{srs.length}"
+
+    miq_servers = srs.select do |svr|
+      svr.vm_scan_storage_affinity? ? all_storage_server_ids.detect { |id| id == svr.id } : storage_server_ids.empty?
+    end
+    _log.debug "miq_servers1.length = #{miq_servers.length}"
+
+    miq_servers.select! do |svr|
+      result = svr.status == "started" && svr.has_zone?(my_zone)
+      result &&= svr.is_vix_disk? if vendor == 'VMware'
```

Comment:
Missed that. I should have spent as much time removing VMware refs from RedHat as I spent removing RedHat refs from VMware.